rust/connlib/clients/shared/src/control.rs

```diff
@@ -301,6 +275,38 @@ impl<CB: Callbacks + 'static> ControlPlane<CB> {
             .send(EgressMessages::CreateLogSink {})
             .await;
     }
+
+    pub async fn handle_tunnel_event(&mut self, event: firezone_tunnel::Event) {
+        match event {
+            firezone_tunnel::Event::SignalIceCandidate { conn_id, candidate } => {
+                let Some(gateway_id) = conn_id.into_gateway_id() else {
+                    return;
+                };
```

Comment:
Needs this on both sides is kind of unfortunate. I think we could probably solve that quite elegantly if `Tunnel` had a type parameter of the "remote ID". In other words, a tunnel on the gateway is parameterized with `ClientId` and vice versa. That would make these things type-safe.

Reply:
We will extract the `IceCandidateGatherer` from `Tunnel` when we replace `webrtc`, right? we can do it then.

Reply:
On a meta note, I am bit worried to fall for the [second-system effect](https://en.wikipedia.org/wiki/Second-system_effect) here where we think "we can fix all these things once we do X".

I think it might be worth it to explore before already, how we can use the type system effectively to help us with using `Tunnel` in its two configurations: on the client side and on the gateway side.